### PR TITLE
Doc language edits - Part 1

### PR DESCRIPTION
**Pull request by yuxizama**
Hey team,

I've read through the first 2 parts of doc and suggested some changes to improve readability. 

Plus, can you add a simple definition of PBS in the "concepts" part? It is referred several times in the "Build-in models" section without any explanation.

Thanks!

### CODE REVIEW

File: docs/built-in-models/linear.md [hunk 1]
```diff
@@ -16,25 +19,24 @@ Concrete ML provides several of the most popular linear models for `regression`
...
+The `n_bits` parameter controls the bit width of the inputs and weights of the linear models. Linear models do not use table lookups and thus alllows weight and inputs to be high precision integers.
 
-The `n_bits` parameter can be set to `8` or more bits for models with up to `300` input dimensions. When the input has more dimensions, `n_bits` must be reduced to `6-7`. All performance metrics are preserved down to `n_bits=6`, compared to the non-quantized float models from scikit-learn.
+For models with input dimensions up to `300`, the parameter `n_bits` can be set to `8` or more. When the input dimensions are larger, `n_bits` must be reduced to `6-7`. Quantized models can preserve all performance metrics compared to the non-quantized float models from scikit-learn when `n_bits` is down to `6`.
```
Comment (andrei-stoian-zama) [inline docs/built-in-models/linear.md:29]:
```suggestion
For models with input dimensions up to `300`, the parameter `n_bits` can be set to `8` or more. When the input dimensions are larger, `n_bits` must be reduced to `6-7`. In many cases, quantized models can preserve all performance metrics compared to the non-quantized float models from scikit-learn when `n_bits` is down to `6`. Users should validate accuracy on held-out test sets and adjust `n_bits` accordingly.
```

File: docs/built-in-models/linear.md [hunk 2]
```diff
@@ -16,25 +19,24 @@ Concrete ML provides several of the most popular linear models for `regression`
...
 
-The same quantization parameters (i.e., scale and zero-point) are applied on all features, so it can be beneficial to make all feature distribution similar by using standard or min-max normalization. For a more detailed comparison of the impact of such pre-processing please refer to [the logistic regression notebook](../advanced_examples/LogisticRegression.ipynb).
+The same quantization parameters such as scale and zero-point are applied on all features, so you should try to make all feature distribution similar by using standard or min-max normalization. For a more detailed comparison of the impact of such pre-processing please refer to [the logistic regression notebook](../advanced_examples/LogisticRegression.ipynb).
```
Comment (andrei-stoian-zama) [inline docs/built-in-models/linear.md:31]:
```suggestion
{% hint style="warning" %}
For best results the distribution of individual features should be similar. This can be achieved using standardization or by using standard or min-max normalization. When many one-hot features are present, [Principal Component Analysis](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html) can be considered as a pre-processing stage. For a more detailed comparison of the impact of such pre-processing please refer to [the logistic regression notebook](../advanced_examples/LogisticRegression.ipynb).
{% endhint %}
```

File: docs/built-in-models/linear.md [hunk 3]
```diff
@@ -16,25 +19,24 @@ Concrete ML provides several of the most popular linear models for `regression`
...
+
+## Pre-trained models
+
+You can convert an already trained scikit-learn linear model to a *Concrete ML one by using the [`from_sklearn_model`](../references/api/concrete.ml.sklearn.base.md#classmethod-from_sklearn_model) method. See [the following example](linear.md#loading-a-pre-trained-model). This functionality is only available for linear models.
```
Comment (andrei-stoian-zama) [inline docs/built-in-models/linear.md:35]:
```suggestion
You can convert an already trained scikit-learn linear model to a *Concrete ML one by using the [`from_sklearn_model`](../references/api/concrete.ml.sklearn.base.md#classmethod-from_sklearn_model) method. See [the following example](linear.md#loading-a-pre-trained-model). 
```

File: docs/built-in-models/linear.md [hunk 4]
```diff
@@ -79,16 +81,18 @@ print(
 # Output:
 #  100 examples over 100 have an FHE inference equal to the clear inference
 ```
+## Model accuracy
+We can plot the decision boundary of the classifier and compare those results with a scikit-learn model executed in clear. You can find the complete code in the [LogisticRegression notebook](../tutorials/ml_examples.md).
```
Comment (andrei-stoian-zama) [inline docs/built-in-models/linear.md:85]:
```suggestion
The figure below shows how to plot the decision boundary of the FHE classifier and compares those results with a scikit-learn model executed in clear. You can find the complete code in the [LogisticRegression notebook](../tutorials/ml_examples.md).
```

File: docs/built-in-models/linear.md [hunk 5]
```diff
@@ -79,16 +81,18 @@ print(
...
+
+The overall accuracy scores are identical (93%) between the scikit-learn model (executed in the clear) and the *Concrete ML one (executed in FHE). In fact, quantization has little impact on the decision boundaries, as linear models can use large precision numbers when quantizing inputs and weights in *Concrete ML. Additionally, as the linear models do not use [Programmable Boostrapping](/getting-started/concepts.md#cryptography-concepts), the FHE computations are always exact. This ensures that the FHE predictions are always identical to the quantized clear ones.
```
Comment (andrei-stoian-zama) [inline docs/built-in-models/linear.md:87]:
```suggestion
The overall accuracy scores are identical (93%) between the scikit-learn model (executed in the clear) and the *Concrete ML one (executed in FHE). In fact, quantization has little impact on the decision boundaries, as linear models can use large precision numbers when quantizing inputs and weights in *Concrete ML. Additionally, as the linear models do not use [Programmable Boostrapping](/getting-started/concepts.md#cryptography-concepts), the FHE computations are always exact, irrespective of the [PBS error tolerance setting](../explanations/advanced_features.md#approximate-computations). This ensures that the FHE predictions are always identical to the quantized clear ones.
```

File: docs/getting-started/concepts.md [hunk 6]
```diff
@@ -1,57 +1,67 @@
...
+- **Quantization**: Concrete ML quantizes inputs, outputs, weights, and activations to meet FHE limitations. See [the quantization documentation]((../explanations/quantization.md)) for details.
 
-Additional specific methods can help to make models compatible with FHE constraints. For instance, dimensionality reduction can reduce the number of input features and, thus, the maximum accumulator bit-width reached within a circuit. Similarly, sparsity-inducing training methods, such as pruning, deactivate some features during inference. For now, dimensionality reduction is considered as a pre-processing step, while pruning is used in the [built-in neural networks](../built-in-models/neural-networks.md).
+   -  **Accuracy trade-off**: Quantization may reduce accuracy, but careful selection can mitigate this. Concrete ML offers built-in quantized models; users only configure parameters like bit width. For more details of quantization configurations, see [the advanced quantization guide](../explanations/quantization.md#configuring-model-quantization-parameters).
```
Comment (andrei-stoian-zama) [inline docs/getting-started/concepts.md:65]:
```suggestion
   -  **Accuracy trade-off**: Quantization may reduce accuracy, but careful selection of quantization parameters or of the training approach can mitigate this. Concrete ML offers built-in quantized models; users only configure parameters like bit width. For more details of quantization configurations, see [the advanced quantization guide](../explanations/quantization.md#configuring-model-quantization-parameters).
```

**Review by andrei-stoian-zama — CHANGES_REQUESTED** [review]
Thanks a lot @yuxizama , just a few suggestions, some based on recent functionality updates

File: docs/built-in-models/linear.md (hunk 1 above)
Comment (yuxizama) [reply docs/built-in-models/linear.md:29]: We can refer to users as "you" => "You should validate accuracy on held-out test sets and adjust `n_bits` accordingly"

File: docs/built-in-models/linear.md (hunk 2 above)
Comment (yuxizama) [reply docs/built-in-models/linear.md:31]: {% hint style="warning" %}
For optimal results, you can use standard or min-max normalization to achieve a similar distribution of individual features. When there are many one-hot features, consider [Principal Component Analysis](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html)  as a pre-processing stage. 

For a more detailed comparison of the impact of such pre-processing,  please refer to [the logistic regression notebook](../advanced_examples/LogisticRegression.ipynb).
{% endhint %}

File: docs/built-in-models/linear.md (hunk 4 above)
Comment (yuxizama) [reply docs/built-in-models/linear.md:85]: The figure below compares the decision boundary of the FHE classifier and a scikit-learn model executed in clear.

File: docs/getting-started/concepts.md [hunk 7]
```diff
@@ -1,57 +1,67 @@
...
 
-To respect FHE constraints, all numerical programs that include non-linear operations over encrypted data must have all inputs, constants, and intermediate values represented with integers of a maximum of 16 bits.
+-  **Key generation:** Cryptographic keys are generated using random number generators. Key generation can be time-consuming and produce large keys, but each model used by a client only requires key generation once.
+
+   - **Private encryption key**: A private encryption key is a series of bits used within an encryption algorithm for encrypting data so that the corresponding ciphertext appears random.
+
+   - **Public evaluation key**: A public evaluation key is used to perform homomorphic operations on encrypted data, typically by a server.
+
+-  **Guaranteed correctness of encrypted computations:** To ensure security, TFHE adds random noise to ciphertexts. Depending on the noise parameters, it can cause errors during encrypted data processing. By default, Concrete ML uses parameters that guarantee the correctness of encrypted computations, so the results on encrypted data equals to those of simulations on clear data.
+- **Programmable Boostrapping (PBS)** : TBD
```
Comment (yuxizama) [inline docs/getting-started/concepts.md:55]: We need a simple explanation of PBS here as a reference for other docs
cc @andrei-stoian-zama

File: docs/built-in-models/linear.md (hunk 1 above)
Comment (andrei-stoian-zama) [reply docs/built-in-models/linear.md:29]: yes, sure

File: docs/getting-started/concepts.md (hunk 7 above)
Comment (andrei-stoian-zama) [reply docs/getting-started/concepts.md:55]: Sure: `programmable bootstrapping enables the homomorphic evaluation of any function of a ciphertext, with a controlled level of noise` (taken from the whitepaper), with a link: https://eprint.iacr.org/2021/091

**Review by andrei-stoian-zama — DISMISSED** [review]
